### PR TITLE
Vagrant

### PR DESCRIPTION
**Pull request by ssayed118**
- Added inputs to Inspec.yml (and several control files) to make the overlays simpler to write
- Updated kitchen files to use a vagrant testing suite

### CODE REVIEW

**Comment by ssayed118** [comment]
Baseline now has all the inputs, and testing each baseline category (mod, mod-hva, etc...) passed tests

**Comment by ejaronne** [comment]
Controls SV-230336 and SV-230374 throw profile errors. Load attached jsons into Heimdall to see that now test runs, but more interestingly, the details tab doesn't completely populate for some reason.
[ssayed_rhel8_vagrant_09132022.json.txt](https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/files/9557614/ssayed_rhel8_vagrant_09132022.json.txt)

**Comment by ejaronne** [comment]
inspec.yml:

Change:

> #SV-230244, SV-244525:
>   - name: mins_of_inactivity
>     desc: Maximum number of minutes of account inactivity before account lockout
>     type: Numeric
>     value: 15
> 
>   - name: system_inactivity_timeout # also used in SV-230353
>     desc: Maximum system inactivity timeout (time in seconds).
>     type: Numeric
>     value: 900
> 
> #SV-230244
>   - name: client_alive_interval 
>     desc: Interval of network connections associated with SSH 
>     type: Numeric
>     value: 1800

to:

#SV-230244, SV-244525:
  - name: mins_of_inactivity
    desc: Maximum number of minutes of account inactivity before account lockout
    type: Numeric
    value: 10

  - name: client_alive_interval 
    desc: Interval of network connections associated with SSH 
    type: Numeric
    value: 600

#SV-230353
  - name: system_inactivity_timeout 
    desc: Maximum system inactivity timeout (time in seconds).
    type: Numeric
    value: 900

**Comment by ejaronne** [comment]
For SV-244525:

Use 
name: mins_of_inactivity
desc: Maximum number of minutes of account inactivity before account lockout
type: Numeric
value: 10

name: client_alive_interval
desc: Interval of network connections associated with SSH
type: Numeric
value: 600

NOT
name: system_inactivity_timeout
which is only for SV-230353

Also: there are additional places in the text not yet using these inputs.

**Comment by ejaronne** [comment]
SV-230331:
The check text reads:
`If any temporary accounts have no expiration date set or do not expire within {""hours""=>72, ""days""=>3} hours, this is a finding.`

instead of:
`If any temporary accounts have no expiration date set or do not expire within 72 hours, this is a finding.`

similar problem for the fix text

**Comment by ejaronne** [comment]
Affects many controls:
change
`unlock_time= #{input('unlock_time')}`
to
`unlock_time=#{input('unlock_time')}`
removing the space after the "=" sign

File: inspec.yml [hunk 1]
```diff
@@ -8,48 +8,130 @@ version: 2.1.2
...
+#SV-230244, SV-244525:
+  - name: mins_of_inactivity
+    desc: Maximum number of minutes of account inactivity before account lockout
+    type: Numeric
+    value: 15
+
+  - name: system_inactivity_timeout # also used in SV-230353
+    desc: Maximum system inactivity timeout (time in seconds).
+    type: Numeric
+    value: 900
```
Comment (karikarshivani) [inline inspec.yml]: I need to take a closer look at this input (which was previously used in SV-244525 but was replaced with `client_alive_interval` later). The only potential use of this input seems to be in SV-230353 which is already covered by `fail_interval`

Comment (ejaronne) [reply inspec.yml]: SV-230353 is a totally separate requirement, and had always used system_inactivity_timeout. I do not see SV-230353 ever using anything else.
https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/blob/fa06c0a620740e98549ffdd4089d2ef3ed7bc490/controls/SV-230353.rb#L52

SV-244525 has only ever used client_alive_interval:
https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/blob/fa06c0a620740e98549ffdd4089d2ef3ed7bc490/controls/SV-244525.rb#L79

**Comment by ejaronne** [comment]
inspec.yml:
known_system_accounts is defined twice. check to make sure inputs are only defined once.

**Comment by ejaronne** [comment]
SV-230229:
replace:
`=#{input('org_name')[:acronym]},`
with
`= #{input('org_name')[:acronym]},`
because the original STIG content has a space after the = sign

**Comment by ejaronne** [comment]
SV-230333:
replace 
    `deny =#{input('unsuccessful_attempts')}`
with
    `deny = #{input('unsuccessful_attempts')}`
because the original STIG content has a space after the = sign

**Comment by ejaronne** [comment]
SV-230335:
replace
    fail_interval =#{input('fail_interval')}

with
    fail_interval = #{input('fail_interval')}

because the original STIG content has a space after the = sign

**Comment by ejaronne** [comment]
SV-230363:

replace
    difok =#{input('difok')}

with
    difok = #{input('difok')}

because the original STIG content has a space after the = sign

**Comment by ejaronne** [comment]
SV-230483:

replace
    space_left =#{input('storage_space_left')}%

with
    space_left = #{input('storage_space_left')}%

because the original STIG content has a space after the = sign

File: inspec.yml (hunk 1 above)
Comment (karikarshivani) [reply inspec.yml]: We're looking at two different control files (and I might be missing context/understanding for this), but looking at the parameterized version of [SV-230353](https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230353.rb), `fail_interval` and `fail_interval_mins` are used and I'm just wondering why we are also using `system_inactivity_timeout` when `fail_interval` has the same value of 900.

If the overlays have different values for `system_inactivity_timeout` and `fail_interval`, then it's totally understandable!

Comment (karikarshivani) [reply inspec.yml]: After our discussion, we have determined to use only the `system_inactivity_timeout` input in [SV-230353](https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230353.rb) and create more inputs as necessary specifically related to `system_inactivity_timeout`

Comment (ejaronne) [reply inspec.yml]: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/pull/1

**Comment by ejaronne** [comment]
wrt https://github.com/CMSgov/cms-ars-5.0-red-hat-enterprise-linux-8-stig-overlay/pull/1#discussion_r978984485
https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230346.rb needs to be parameterized using the maxlogins_limit input.
Also: "Consider making the data type integer in the baseline (log an issue if short on time)"

**Comment by ejaronne** [comment]
parameterize title, description, check, fix text for SV-230369, SV-230370 using input('min_len')

**Comment by ejaronne** [comment]
SV-230359 is N/A for HVA baselines

**Comment by karikarshivani** [comment]
> wrt [CMSgov/cms-ars-5.0-red-hat-enterprise-linux-8-stig-overlay#1 (comment)](https://github.com/CMSgov/cms-ars-5.0-red-hat-enterprise-linux-8-stig-overlay/pull/1#discussion_r978984485) https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/blob/vagrant/controls/SV-230346.rb needs to be parameterized using the maxlogins_limit input. Also: "Consider making the data type integer in the baseline (log an issue if short on time)"

[Issue logged](https://github.com/CMSgov/redhat-enterprise-linux-8-stig-baseline/issues/8) for data type conversion

**Comment by ejaronne** [comment]
SV-230352 should be parameterized based on the input "system_inactivity_timeout", similar to SV-230353

**Comment by ejaronne** [comment]
- SV-230225: parameterize entire banner content in Desc, Check, and Fix text using input: banner_message_text_ral
- SV-230226: parameterize entire banner content in Desc, Check, and Fix text using input: banner_message_text_gui
- SV-230227: parameterize entire banner content in Desc, Check, and Fix text using input: banner_message_text_cli

File: controls/SV-230335.rb [hunk 2]
```diff
@@ -24,23 +24,23 @@
...
+    fail_interval = #{input('fail_interval')}
 
-    If the \"fail_interval\" option is not set to \"900\" or more, is missing
+    If the \"fail_interval\" option is not set to \"#{input('fail_interval')}\" or more, is missing
 or commented out, this is a finding.
   "
   desc  'fix', "
-    Configure the operating system to lock an account when three unsuccessful
+    Configure the operating system to lock an account when #{input('unsuccessful_attempts')} unsuccessful
 logon attempts occur in 15 minutes.
```
Comment (karikarshivani) [inline controls/SV-230335.rb]: Parameterize 15 minutes using `#{input('fail_interval')/60}`

File: controls/SV-230342.rb [hunk 3]
```diff
@@ -37,8 +37,8 @@
     $ sudo grep pam_faillock.so /etc/pam.d/password-auth
```
Comment (karikarshivani) [inline controls/SV-230342.rb:37]: **Note** paragraph in check text needs to be parameterized for `three` and `15`

File: controls/SV-230466.rb [hunk 4]
```diff
@@ -12,7 +12,7 @@
 generated. This set of events is typically a subset of the list of all events
 for which the system is capable of generating audit records.
 
-    DoD has defined the list of events for which RHEL 8 will provide an audit
+  #{input('org_name')[:acronym]} has defined the list of events for which RHEL 8 will provide an audit
```
Comment (karikarshivani) [inline controls/SV-230466.rb:15]: This control has check text that needs to be parameterized for unsuccessful logon attempts and fail interval.

File: controls/SV-230335.rb (hunk 2 above)
Comment (ejaronne) [reply controls/SV-230335.rb]: fixed: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/commit/c1129015b32e7c89b56ac7b21ded3b310f92185f

File: controls/SV-230342.rb (hunk 3 above)
Comment (ejaronne) [reply controls/SV-230342.rb:37]: fixed: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/commit/082f8a8681349c96bf8cb5667ede4874a8697b28

File: controls/SV-230466.rb (hunk 4 above)
Comment (ejaronne) [reply controls/SV-230466.rb:15]: fixed: https://github.com/ssayed118/redhat-enterprise-linux-8-stig-baseline/commit/639f9115af83a3f73e368c486fe8af1d20e7a912

**Review by ejaronne — APPROVED** [review]
Looks good.

**Pull request merged**